WordPress/Classes/Services/BlogJetpackSettingsService.swift

```diff
@@ -70,55 +70,28 @@ struct BlogJetpackSettingsService {
-                                    key: BlogJetpackSettingsServiceRemote.Keys.monitorEnabled,
-                                    value: value as AnyObject,
-                                    success: success,
-                                    failure: failure)
-    }
-
-    func updateBlockMaliciousLoginAttemptsForBlog(_ blog: Blog, value: Bool, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
-        updateJetpackSettingForBlog(blog,
-                                    key: BlogJetpackSettingsServiceRemote.Keys.blockMaliciousLoginAttempts,
-                                    value: value as AnyObject,
-                                    success: success,
-                                    failure: failure)
-    }
-
-    func updateWhiteListedIPAddressesForBlog(_ blog: Blog, value: Set<String>, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
-        let joinedIPs = value.joined(separator: ", ")
-        updateJetpackSettingForBlog(blog,
-                                    key: BlogJetpackSettingsServiceRemote.Keys.whiteListedIPAddresses,
-                                    value: joinedIPs as AnyObject,
-                                    success: success,
-                                    failure: failure)
-    }
-
-    func updateSSOEnabledForBlog(_ blog: Blog, value: Bool, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
-        updateJetpackSettingForBlog(blog,
-                                    key: BlogJetpackSettingsServiceRemote.Keys.ssoEnabled,
-                                    value: value as AnyObject,
-                                    success: success,
-                                    failure: failure)
-    }
+    func updateJetpackSettingsForBlog(_ blog: Blog, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
+        guard let remote = BlogJetpackSettingsServiceRemote(wordPressComRestApi: blog.wordPressComRestApi()),
+            let blogDotComId = blog.dotComID as? Int,
+            let blogSettings = blog.settings else {
+                failure(nil)
+                return
+        }
 
-    func updateSSOMatchAccountsByEmailForBlog(_ blog: Blog, value: Bool, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
-        updateJetpackSettingForBlog(blog,
-                                    key: BlogJetpackSettingsServiceRemote.Keys.ssoMatchAccountsByEmail,
-                                    value: value as AnyObject,
-                                    success: success,
-                                    failure: failure)
-    }
+        remote.updateJetpackSettingsForSite(blogDotComId,
+                                            settings: jetpackSettingsRemote(blogSettings),
+                                            success: {
+                                                do {
+                                                    try self.context.save()
```

Comment:
Now that we can do batch updates, should the UI update immediately? For instance, if you disable brute force prevention, the whitelist option only disappears after the save is completed.